<a href="https://colab.research.google.com/github/CodeDJR/Chipotle-dataset-Data-Cleaning-using-Pandas/blob/main/week3_set2_Daniel_DAB3096.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**IMPORTING THE PACKAGES**

In [ ]:
import pandas as pd
import numpy as np

**VARIABLE ASSIGNMENT**

In [ ]:
chipotle = pd.read_csv('/content/drive/MyDrive/chipotle.tsv', sep='\t')

# **DATASET SHAPE**

In [ ]:
# Checking the shape of the dataset
chipotle.shape

(4622, 5)

The given dataset has 4622 rowas and 5 columns

# **DATATYPES**

In [ ]:
# Checking the datatypes of each columns
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


# **MISSING VALUES**

In [ ]:
chipotle.isna().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

We have only one column with 1246 empty cells which is the column named 'choice_description'.

# **DUPLICATED VALUES**

In [ ]:
# checking the number of duplicate rows
duplicates_count = chipotle.duplicated().sum()
print(duplicates_count)

59


In [ ]:
# viewing the duplicate rows
chipotle[chipotle.duplicated()]

order_id  quantity                            item_name  \
238        103         1                        Steak Burrito   
248        108         1                          Canned Soda   
297        129         1                        Steak Burrito   
381        165         1                    Canned Soft Drink   
484        205         1                         Chicken Bowl   
567        233         1                    Canned Soft Drink   
617        254         1                                Chips   
684        282         1                    Canned Soft Drink   
773        319         1                         Chicken Bowl   
908        376         1                        Steak Burrito   
924        381         1                   Chicken Soft Tacos   
1003       412         1                        Bottled Water   
1117       462         1                      Chicken Burrito   
1203       491         1                      Chicken Burrito   
1245       507         1                    Canned Soft Drink   
1381       561         1                    Canned Soft Drink   
1410       573         1                        Bottled Water   
1420       576         1                        Barbacoa Bowl   
1421       576         1                  Barbacoa Salad Bowl   
1502       610         1                      Chicken Burrito   
1534       624         1  Chips and Tomatillo-Red Chili Salsa   
1548       628         1                  Chips and Guacamole   
1612       651         1                                 Izze   
1791       724         1                    Canned Soft Drink   
1792       724         1                    Canned Soft Drink   
2078       838         1                   Chicken Soft Tacos   
2127       856         1                      Chicken Burrito   
2307       926         1                         Chicken Bowl   
2318       926         1                         Chicken Bowl   
2362       940         1                    Canned Soft Drink   
2448       972         1                         Chicken Bowl   
2553      1011         1                    Canned Soft Drink   
2554      1011         1                    Canned Soft Drink   
2560      1013         1                  Chips and Guacamole   
2576      1020         1  Chips and Tomatillo Red Chili Salsa   
2662      1057         1                   Chicken Soft Tacos   
2708      1074         1                                Chips   
2717      1078         1                         Chicken Bowl   
2833      1125         1                      Chicken Burrito   
2881      1144         1                        Bottled Water   
2973      1182         1                         Chicken Bowl   
2977      1184         1                           Steak Bowl   
3121      1244         1                  Chips and Guacamole   
3430      1377         1  Chips and Tomatillo Red Chili Salsa   
3637      1454         1                   Chicken Soft Tacos   
3716      1484         1                          Canned Soda   
3764      1507         1                      Chicken Burrito   
3829      1534         1                      Chicken Burrito   
3924      1574         1                        Bottled Water   
4112      1647         1                         Chicken Bowl   
4155      1660         1                  Chips and Guacamole   
4236      1691         1                         Chicken Bowl   
4266      1701         1                                Chips   
4372      1745         1                  Chips and Guacamole   
4463      1778         1                    Canned Soft Drink   
4479      1785         1                        Bottled Water   
4544      1806         1                        Bottled Water   
4570      1817         1                        Bottled Water   
4603      1827         1                     Barbacoa Burrito   

                                     choice_description item_price  
238   [Tomatillo Red Chili Salsa, [R

In [ ]:
chipotle.drop_duplicates()

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4563 rows x 5 columns]

Now the shape of the dataset is 4563 rows × 5 columns

#**QUANTITY AND ITEM PRICE**

In [ ]:
print(chipotle['quantity'].describe())

count    4622.000000
mean        1.075725
std         0.410186
min         1.000000
25%         1.000000
50%         1.000000
75%         1.000000
max        15.000000
Name: quantity, dtype: float64


In [ ]:
print(chipotle['item_price'].describe())

count       4622
unique        78
top       $8.75 
freq         730
Name: item_price, dtype: object


In [ ]:
# changing the values in 'item-price' to complete numeric value.
chipotle['item_price'] = chipotle['item_price'].str.replace('$','').astype(float)

In [ ]:
chipotle['item_price']

0        2.39
1        3.39
2        3.39
3        2.39
4       16.98
        ...  
4617    11.75
4618    11.75
4619    11.25
4620     8.75
4621     8.75
Name: item_price, Length: 4622, dtype: float64

In [ ]:
chipotle.head(10)
# Now the item_price column is of the type 'float'

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
6                                                NaN        1.69  
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.75  
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...        9.25  
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...        9.25

In [ ]:
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   int64  
 2   item_name           4622 non-null   object 
 3   choice_description  3376 non-null   object 
 4   item_price          4622 non-null   float64
dtypes: float64(1), int64(2), object(2)
memory usage: 180.7+ KB


#  **CHOICE DESCRIPTION**

In [ ]:
# ANALYSING CHOICE DESCRIPTION COLUMN
# NUMBER OF EMPTY VALUES
chipotle['choice_description'].isna().sum()

1246

In [ ]:
chipotle['item_name'].value_counts()

item_name
Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48


In [ ]:
chipotle['choice_description'].value_counts()

choice_description
[Diet Coke]                                                                                                                                      134
[Coke]                                                                                                                                           123
[Sprite]                                                                                                                                          77
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]                                                                            42
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Guacamole, Lettuce]]                                                                 40
                                                                                                                                                ... 
[Fresh Tomato Salsa (Mild), [Pinto Beans, Black Beans, Rice, Cheese, Sour Cream, Lettuc

In [ ]:
chipotle.groupby('item_name').sum().sort_values(['quantity'], ascending=False)

order_id  quantity  \
item_name                                                   
Chicken Bowl                             713926       761   
Chicken Burrito                          497303       591   
Chips and Guacamole                      449959       506   
Steak Burrito                            328437       386   
Canned Soft Drink                        304753       351   
Chips                                    208004       230   
Steak Bowl                               193752       221   
Bottled Water                            175944       211   
Chips and Fresh Tomato Salsa             100419       130   
Canned Soda                               76396       126   
Chicken Salad Bowl                       117104       123   
Chicken Soft Tacos                        98395       120   
Side of Chips                             84769       110   
Veggie Burrito                            80962        97   
Barbacoa Burrito                          74718        91   
Veggie Bowl                               72270        87   
Carnitas Bowl                             62742        71   
Barbacoa Bowl                             53972        66   
Carnitas Burrito                          51652        60   
Steak Soft Tacos                          50177        56   
6 Pack Soft Drink                         52322        55   
Chips and Tomatillo Red Chili Salsa       45556        50   
Chicken Crispy Tacos                      47382        50   
Chips and Tomatillo Green Chili Salsa     41170        45   
Carnitas Soft Tacos                       37673        40   
Steak Crispy Tacos                        35818        36   
Chips and Tomatillo-Green Chili Salsa     21826        33   
Steak Salad Bowl                          29590        31   
Nantucket Nectar                          21814        29   
Chips and Tomatillo-Red Chili Salsa       18344        25   
Barbacoa Soft Tacos                       18725        25   
Chips and Roasted Chili Corn Salsa        14999        23   
Izze                                      10587        20   
Veggie Salad Bowl                         17303        18   
Chips and Roasted Chili-Corn Salsa        17462        18   
Barbacoa Crispy Tacos                      5613        12   
Barbacoa Salad Bowl                        9708        10   
Chicken Salad                             10465         9   
Carnitas Crispy Tacos                      5868         8   
Veggie Soft Tacos                          5520         8   
Burrito                                    1550         6   
Veggie Salad                               7265         6   
Carnitas Salad Bowl                        6693         6   
Bowl                                        472         4   
Steak Salad                                2362         4   
Salad                                      1150         2   
Crispy Tacos                                434         2   
Chips and Mild Fresh Tomato Salsa           279         1   
Carnitas Salad                             1500         1   
Veggie Crispy Tacos                         668         1   

                                                                      choice_description  \
item_name                                                                                  
Chicken Bowl                           [Tomatillo-Red Chili Salsa (Hot), [Black Beans...   
Chicken Burrito                        [Tomatillo-Green Chili Salsa (Medium), [Pinto ...   
Chips and Guacamole                                                                    0   
Steak Burrito                          [Tomatillo Red Chili Salsa, [Fajita Vegetables...   
Canned Soft Drink                      [Coke][Sprite][Coke][Coke][Lemonade][Sprite][D...   
Chips                                                                                  0   
Steak Bowl                             [Fresh Tomato Salsa (Mild), [Black Beans, Rice...   
Bottled Water                         

In [ ]:
chipotle['choice_description'] = chipotle['choice_description'].fillna('single choice')   # handling missing values
chipotle['choice_description']

0                                           single choice
1                                            [Clementine]
2                                                 [Apple]
3                                           single choice
4       [Tomatillo-Red Chili Salsa (Hot), [Black Beans...
                              ...                        
4617    [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...
4618    [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...
4619    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...
4620    [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...
4621    [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...
Name: choice_description, Length: 4622, dtype: object

In [ ]:
chipotle.isna().sum()

order_id              0
quantity              0
item_name             0
choice_description    0
item_price            0
dtype: int64

In [ ]:
chipotle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   int64  
 2   item_name           4622 non-null   object 
 3   choice_description  4622 non-null   object 
 4   item_price          4622 non-null   float64
dtypes: float64(1), int64(2), object(2)
memory usage: 180.7+ KB


# **HANDLING SPECIAL CHARACTERS IN TEXT BASED COLUMNS**

In [ ]:
import re

In [ ]:
def has_special_characters(s):
    # This regex matches any character that is not a word character or whitespace
    return bool(re.search(r'[^\w\s]', s))
chipotle['Has Special Characters'] = chipotle['item_name'].apply(has_special_characters)


In [ ]:
special_chars_df = chipotle[chipotle['Has Special Characters']]
print(special_chars_df)

      order_id  quantity                              item_name  \
3            1         1  Chips and Tomatillo-Green Chili Salsa   
15           8         1  Chips and Tomatillo-Green Chili Salsa   
30          15         1  Chips and Tomatillo-Green Chili Salsa   
113         50         1  Chips and Tomatillo-Green Chili Salsa   
133         58         1  Chips and Tomatillo-Green Chili Salsa   
...        ...       ...                                    ...   
3810      1527         1  Chips and Tomatillo-Green Chili Salsa   
3871      1552         1  Chips and Tomatillo-Green Chili Salsa   
3905      1565         1     Chips and Roasted Chili-Corn Salsa   
4007      1604         1  Chips and Tomatillo-Green Chili Salsa   
4088      1637         1    Chips and Tomatillo-Red Chili Salsa   

     choice_description  item_price  Has Special Characters  
3         single choice        2.39                    True  
15        single choice        2.39                    True  
30       

In [ ]:
chipotle['item_name'] = chipotle['item_name'].str.replace(r'[^\w\s]', ' ', regex=True)

In [ ]:
chipotle['item_name'].value_counts()

item_name
Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Chips and Tomatillo Green Chili Salsa     74
Carnitas Bowl                             68
Chips and Tomatillo Red Chili Salsa       68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55


In [ ]:
chipotle['choice_description'] = chipotle['choice_description'].str.replace(r'[^\w\s,]', '', regex=True)

In [ ]:
chipotle['choice_description'].value_counts()

choice_description
single choice                                                                                                                       1246
Diet Coke                                                                                                                            134
Coke                                                                                                                                 123
Sprite                                                                                                                                77
Fresh Tomato Salsa, Rice, Black Beans, Cheese, Sour Cream, Lettuce                                                                    42
                                                                                                                                    ... 
Fresh Tomato Salsa Mild, Pinto Beans, Black Beans, Rice, Cheese, Sour Cream, Lettuce                                                   1
Fresh Tomato Salsa Mil

In [ ]:
chipotle.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description  item_price  \
0                                      single choice        2.39   
1                                         Clementine        3.39   
2                                              Apple        3.39   
3                                      single choice        2.39   
4  TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
5  Fresh Tomato Salsa Mild, Rice, Cheese, Sour Cr...       10.98   
6                                      single choice        1.69   
7  Tomatillo Red Chili Salsa, Fajita Vegetables, ...       11.75   
8  Tomatillo Green Chili Salsa, Pinto Beans, Chee...        9.25   
9  Fresh Tomato Salsa, Rice, Black Beans, Pinto B...        9.25   

   Has Special Characters  
0                   False  
1                   False  
2                   False  
3                    True  
4                   False  
5                   False  
6                   False  
7                   False  
8                   False  
9                   False

In [ ]:
chipotle.drop('Has Special Characters', axis =1)

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                         single choice        2.39  
1                                            Clementine        3.39  
2                                                 Apple        3.39  
3                                         single choice        2.39  
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98  
...                                                 ...         ...  
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75  
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75  
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25  
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75  
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75  

[4622 rows x 5 columns]

Finally removed the speacial characters in the text based columns for consistency and data integrity.

# **ITEM NAME STANDARDIZATION**

In [ ]:
chipotle.head(10)

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description  item_price  \
0                                      single choice        2.39   
1                                         Clementine        3.39   
2                                              Apple        3.39   
3                                      single choice        2.39   
4  TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
5  Fresh Tomato Salsa Mild, Rice, Cheese, Sour Cr...       10.98   
6                                      single choice        1.69   
7  Tomatillo Red Chili Salsa, Fajita Vegetables, ...       11.75   
8  Tomatillo Green Chili Salsa, Pinto Beans, Chee...        9.25   
9  Fresh Tomato Salsa, Rice, Black Beans, Pinto B...        9.25   

   Has Special Characters  
0                   False  
1                   False  
2                   False  
3                    True  
4                   False  
5                   False  
6                   False  
7                   False  
8                   False  
9                   False

All the item names are in consistent with respect to case nature.

In [ ]:
chipotle

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                         single choice        2.39   
1                                            Clementine        3.39   
2                                                 Apple        3.39   
3                                         single choice        2.39   
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
...                                                 ...         ...   
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75   
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75   
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25   
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75   
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75   

      Has Special Characters  
0                      False  
1                      False  
2                      False  
3                       True  
4                      False  
...                      ...  
4617                   False  
4618                   False  
4619                   False  
4620                   False  
4621                   False  

[4622 rows x 6 columns]

In [ ]:
chipotle

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                         single choice        2.39   
1                                            Clementine        3.39   
2                                                 Apple        3.39   
3                                         single choice        2.39   
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
...                                                 ...         ...   
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75   
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75   
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25   
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75   
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75   

      Has Special Characters  
0                      False  
1                      False  
2                      False  
3                       True  
4                      False  
...                      ...  
4617                   False  
4618                   False  
4619                   False  
4620                   False  
4621                   False  

[4622 rows x 6 columns]

In [ ]:
chipotle.drop('Has Special Characters', axis = 1)

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                         single choice        2.39  
1                                            Clementine        3.39  
2                                                 Apple        3.39  
3                                         single choice        2.39  
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98  
...                                                 ...         ...  
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75  
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75  
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25  
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75  
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75  

[4622 rows x 5 columns]

# **QUESTIONS**

1). Quantity of each item?

In [ ]:
chipotle['item_name'].value_counts()

item_name
Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chips and Fresh Tomato Salsa             110
Chicken Salad Bowl                       110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Chips and Tomatillo Green Chili Salsa     74
Carnitas Bowl                             68
Chips and Tomatillo Red Chili Salsa       68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55


2). What was the most ordered item in the choice_description column?

In [ ]:
chipotle['choice_description'].value_counts()

choice_description
single choice                                                                                                                       1246
Diet Coke                                                                                                                            134
Coke                                                                                                                                 123
Sprite                                                                                                                                77
Fresh Tomato Salsa, Rice, Black Beans, Cheese, Sour Cream, Lettuce                                                                    42
                                                                                                                                    ... 
Fresh Tomato Salsa Mild, Pinto Beans, Black Beans, Rice, Cheese, Sour Cream, Lettuce                                                   1
Fresh Tomato Salsa Mil

**The most ordered item in the choice description is 'Diet Coke'**

3). How many items were ordered in total?

In [ ]:
chipotle['quantity'].sum()

4972

4). How much was the revenue for the period in the dataset?

In [ ]:
Revenue = sum(chipotle['item_price']*chipotle['quantity'])
Revenue

39237.020000000055

In [ ]:
chipotle

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                         single choice        2.39   
1                                            Clementine        3.39   
2                                                 Apple        3.39   
3                                         single choice        2.39   
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
...                                                 ...         ...   
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75   
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75   
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25   
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75   
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75   

      Has Special Characters  
0                      False  
1                      False  
2                      False  
3                       True  
4                      False  
...                      ...  
4617                   False  
4618                   False  
4619                   False  
4620                   False  
4621                   False  

[4622 rows x 6 columns]

5). How many orders were made in the period?

In [ ]:
chipotle['order_id'].nunique()

1834

7). Number of different item sold.

In [ ]:
chipotle['item_name'].nunique()

47

So totally 47 different items were sold

6). What is the average revenue amount per order?





In [ ]:
chipotle['revenue'] = chipotle['quantity'] * chipotle['item_price']

In [ ]:
chipotle

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  \
0                                         single choice        2.39   
1                                            Clementine        3.39   
2                                                 Apple        3.39   
3                                         single choice        2.39   
4     TomatilloRed Chili Salsa Hot, Black Beans, Ric...       16.98   
...                                                 ...         ...   
4617  Fresh Tomato Salsa, Rice, Black Beans, Sour Cr...       11.75   
4618  Fresh Tomato Salsa, Rice, Sour Cream, Cheese, ...       11.75   
4619  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...       11.25   
4620     Fresh Tomato Salsa, Fajita Vegetables, Lettuce        8.75   
4621  Fresh Tomato Salsa, Fajita Vegetables, Pinto B...        8.75   

      Has Special Characters  revenue  
0                      False     2.39  
1                      False     3.39  
2                      False     3.39  
3                       True     2.39  
4                      False    33.96  
...                      ...      ...  
4617                   False    11.75  
4618                   False    11.75  
4619                   False    11.25  
4620                   False     8.75  
4621                   False     8.75  

[4622 rows x 7 columns]

In [ ]:
# Calculation revenue for each order id.
order_revenue = chipotle.groupby('order_id')['revenue'].sum()
order_revenue

order_id
1       11.56
2       33.96
3       12.67
4       21.00
5       13.70
        ...  
1830    23.00
1831    12.90
1832    13.20
1833    23.50
1834    28.75
Name: revenue, Length: 1834, dtype: float64

In [ ]:
# Now calculating the average for each order.
average_order_revenue = order_revenue.mean()
average_order_revenue

21.39423118865867